ui/pages/confirmation/confirmation.js

```diff
@@ -256,20 +260,37 @@ export default function ConfirmationPage({
     // viewed index, reset the index.
     if (
       pendingConfirmations.length === 0 &&
+      pendingApprovalFlows.length === 0 &&
       redirectToHomeOnZeroConfirmations
     ) {
       history.push(DEFAULT_ROUTE);
-    } else if (pendingConfirmations.length <= currentPendingConfirmation) {
+    } else if (
+      pendingConfirmations.length &&
+      pendingConfirmations.length <= currentPendingConfirmation
+    ) {
       setCurrentPendingConfirmation(pendingConfirmations.length - 1);
     }
   }, [
     pendingConfirmations,
+    pendingApprovalFlows,
     history,
     currentPendingConfirmation,
     redirectToHomeOnZeroConfirmations,
   ]);
 
+  useEffect(() => {
+    setApprovalFlowLoadingText(
+      pendingApprovalFlows.length > 0
+        ? pendingApprovalFlows[pendingApprovalFlows.length - 1].loadingText
+        : null,
+    );
+  }, [pendingApprovalFlows]);
+
   if (!pendingConfirmation) {
+    if (pendingApprovalFlows.length > 0) {
+      return <Loading loadingMessage={t(approvalFlowLoadingText)} />;
```

Comment:
Did we decide that we'd require the caller to translate if they wished meaning we can safely assume this is raw text to be rendered?

I understood we didn't want to use a dynamic translation reference as we try to keep them hard coded so we can monitor which translations are in use.

Reply:
I'm not sure what's the best approach. We can either treat it as raw text (maybe translated in the backend if necessary) or treat it as it needing translation. It's just a matter of removing `t`.